.github/workflows/ci.yml

```diff
@@ -114,3 +114,27 @@ jobs:
           else
             exit 0
           fi
+  doxygen:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - name: Install Doxygen
+        run: |
+          wget -qO- "http://doxygen.nl/files/doxygen-1.8.20.linux.bin.tar.gz" | sudo tar --strip-components=1 -xz -C /usr/local
+          sudo apt-get install -y libclang-9-dev
+      - name: Run Doxygen on all library submodules
+        run: |
+          ROOT=$PWD
+          for lib in $(find libraries/aws libraries/standard -maxdepth 1 -mindepth 1)
+          do
+            cd $lib
+            if [ ! -d docs/doxygen/ ]; then exit 0; fi
```

Comment:
Should we log a message if there is nothing in `docs/doxygen`?

Reply:
I can see that being useful I will add that